frontend/src/app/game/game.service.ts

```diff
@@ -233,4 +233,10 @@ export class GameService {
       if (this.tickerId) this.scheduleTick(this.speedInterval);
     }, this.speedInterval);
   }
+
+  pause() {
+    if (this.speed === 'pause') return;
```

Comment:
This row is little bit redudant bcs same condition is on first line in `this.setSpeed()`. But no big deal :)